app/assets/javascripts/discourse/app/controllers/full-page-search.js

```diff
@@ -31,24 +43,37 @@ export default Controller.extend({
   bulkSelectEnabled: null,
 
   loading: false,
-  queryParams: ["q", "expanded", "context_id", "context", "skip_context"],
-  q: null,
+  queryParams: [
+    "q",
+    "expanded",
+    "context_id",
+    "context",
+    "skip_context",
+    "search_type",
+  ],
+  q: undefined,
   selected: [],
```

Comment:
I know it's not your code but this array initialization is incorrect and should be moved to `init()`. Could you do that since you're in there anyway? Thanks!